librz/core/cmd_system.c

```diff
@@ -0,0 +1,217 @@
+static char *config_path(RzCore *core) {
+	char *path = NULL;
+	int fd = rz_file_mkstemp(NULL, &path);
+	if (fd >= 0) {
+		close(fd);
+	}
+	if (!path) {
+		return NULL;
+	}
+
+	Sdb *sdb = sdb_new(NULL, path, 0);
+	if (!sdb) {
+		return NULL;
+	}
+	rz_config_serialize(core->config, sdb);
+	sdb_sync(sdb);
+	sdb_free(sdb);
+
+	return path;
+}
+
+static const char *system_apply_env_var(const char *env, const char *value, const char *arg, RzList *alloc) {
+	char *string = NULL;
+	RzListIter *it = NULL;
+	size_t len = strlen(arg);
+	if (!strstr(arg, env)) {
+		return arg;
+	} else if (strlen(env) == len) {
+		return value;
+	}
+
+	if ((it = rz_list_find_ptr(alloc, (void *)arg))) {
+		string = it->data;
+	} else {
+		string = strdup(arg);
+	}
+
```

Comment:
I don't see what advantages `alloc` array would bring. I would just go with `rz_str_replace` and remove `alloc`. Moreover, you can only find the same `arg` in `alloc` only if it's the same argv[i], so probably a list is not really necessary anyway.

Reply:
alloc, is just a simple idea where i place the allocated strings and just free them via rz_list so i am sure that i never forget to free an allocated value, so i do not have to care whether argv is allocated on the heap or not.